server/jetstream_cluster.go

```diff
@@ -1001,10 +1013,10 @@ func (js *jetStream) applyStreamEntries(mset *Stream, ce *CommittedEntry) (bool,
 				if err != nil {
 					panic(err.Error())
 				}
-				if lseq == 0 && mset.lastSeq() != 0 { // Very first msg
+				// Skip by hand here since first msg special case.
```

Comment:
Not sure I fully understand this comment :-)

Reply:
ok will make clearer. sequences are unsigned, and each message we replicate has the prior sequence it expects when being placed in the stream storage. So 0 can't have prev of -1.